src/main/kotlin/org/jmailen/gradle/kotlinter/support/RuleSets.kt

```diff
@@ -1,29 +1,29 @@
 package org.jmailen.gradle.kotlinter.support
 
-import com.pinterest.ktlint.core.RuleSet
-import com.pinterest.ktlint.core.RuleSetProvider
+import com.pinterest.ktlint.core.RuleProvider
+import com.pinterest.ktlint.core.RuleSetProviderV2
 import com.pinterest.ktlint.ruleset.experimental.ExperimentalRuleSetProvider
 import java.util.ServiceLoader
-import kotlin.comparisons.compareBy
 
-fun resolveRuleSets(
-    providers: Iterable<RuleSetProvider>,
+internal fun resolveRuleProviders(
+    providers: Iterable<RuleSetProviderV2>,
     includeExperimentalRules: Boolean = false,
-): List<RuleSet> {
-    return providers
-        .filter { includeExperimentalRules || it !is ExperimentalRuleSetProvider }
-        .map { it.get() }
-        .sortedWith(
-            compareBy {
-                when (it.id) {
-                    "standard" -> 0
-                    else -> 1
-                }
-            },
-        )
-}
+): Set<RuleProvider> = providers
+    .asSequence()
+    .filter { includeExperimentalRules || it !is ExperimentalRuleSetProvider }
+    .sortedWith(
+        compareBy {
+            when (it.id) {
+                "standard" -> 0
+                else -> 1
+            }
+        },
+    )
```

Comment:
I didn't understand why sorting was required, but I preserved the behavior, despite the new returned type is a `Set`

Reply:
At one point way back we intentionally put standard rules first in the list although I can't dig up the reason why. It might have been to report them first, but this might be an obsolete concern.